src/Confluent.Kafka/IConsumer.cs

```diff
@@ -70,9 +64,9 @@ public interface IConsumer<TKey, TValue> : IClient
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey,TValue}.Assignment" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey,TValue}.AssignedPartitions" />
         /// </summary>
-        List<TopicPartition> Assignment { get; }
+        List<TopicPartition> AssignedPartitions { get; }
```

Comment:
I dont see why the .net client should deviate from the other clients with its own nomenclature.
consistency and familiarity is important

Reply:
Leaving this property as `Assignment` would negate the whole point of this PR - to allow what's going on to be explained without ambiguity. A property called `Assignment` would imply the set of partitions in the consumer group assignment, not the set of partitions assigned to be read from.

Reply:
On second thoughts, I think  you're right - use of the word `Assignment` here is probably ok. The super important thing to get right is the same word shouldn't be overloaded to mean two different things in the api, and the rebalance handler name change fixes that (by avoiding the need to refer to the consumer group assignment set in the api).

It would be ideal to have different terms for the set of partitions in a consumer group assignment and the set of partitions assigned to - this would make documentation easier / clearer. The only viable thing I could think of previously was to consider assign/assigned and assignment to be two different things. But this is a clutching at straws really and isn't much better than really wordy explanations. I'll change back to `Assignment`, and resign myself to really wordy explanations.

Reply:
Sure, there is the assignment from the group leader, and there is the assigned partitions to consume.
In practice they're almost always the same, and the time window where they are different things is about the first couple of lines in the rebalance handler, so I don't think we need to overly educate our users with the difference.

Reply:
I disagree, but at least we have an api now that we can both tolerate.